### PR TITLE
[bugfix] DP accuracy issue fix

### PR DESCRIPTION
**Pull request by ganyi1996ppo**
### What this PR does / why we need it?

When dummy run enabled on data parallel, we assume the non-received kv cache and hidden_states as dummy run dispatched from dp proxy, for those case, we just adopt random empty tensor as its hidden_states and kv_cache and bypass it by default to prevent the stuck on collective call over dp rank.

### CODE REVIEW

File: vllm_ascend/distributed/kv_transfer/simple_connector.py [hunk 1]
```diff
@@ -366,7 +382,12 @@ def recv_kv_caches_and_hidden_states(
             else:
                 hidden_or_intermediate_states = torch.cat(
                     hidden_or_intermediate_states_for_one_req, dim=0)
-
+        # when dummy run is not enabled and some of the kv cache also missed.
+        else:
+            logger.warning(
+                "[rank%d]: Failed to receive all KVs and hidden "
+                "states, redo model forwarding.", torch.distributed.get_rank())
```
Comment (wangxiyuan) [inline vllm_ascend/distributed/kv_transfer/simple_connector.py]: re-do

**Review by wangxiyuan — APPROVED** [review]

File: vllm_ascend/distributed/kv_transfer/simple_connector.py (hunk 1 above)
Comment (ganyi1996ppo) [reply vllm_ascend/distributed/kv_transfer/simple_connector.py]: done
